Add Gaudi inference scheduling CI workflow test

yuanwu2017 opened the pull request:
Gaudi inference scheduling example has been merged. This patch is for enabling the CI workflow test for Gaudi inference scheduling.
- Support manual trigger via workflow_dispatch
- Auto trigger on yaml file changes:
  * values-gaudi.yaml
  * gaie-inference-scheduling directory
  * helmfile.yaml.gotmpl
- Run on gaudi hardware
- Deploy and validate inference scheduling with Gaudi
- Collect logs and cleanup resources

yuanwu2017 committed:
Add Gaudi inference scheduling CI workflow test

- Support manual trigger via workflow_dispatch
- Auto trigger on yaml file changes:
  * values-gaudi.yaml
  * gaie-inference-scheduling directory
  * helmfile.yaml.gotmpl
- Run on gaudi hardware
- Deploy and validate inference scheduling with Gaudi
- Collect logs and cleanup resources

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
Add workflow file itself to trigger paths

This allows the workflow to run when the workflow file is modified in PRs

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
Remove push trigger to avoid duplicate runs in PRs

Keep only pull_request and workflow_dispatch triggers

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
Run on hpu

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
Improve error handling and logging in Gaudi workflow

- Add exit code check for helmfile deployment
- Display full deployment log on failure
- Add warning message for missing helm releases
- Show helm list after successful deployment

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
Add docs

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
Use 1.34

Signed-off-by: yuanwu <yuan.wu@intel.com>

poussa commented on guides/inference-scheduling/ms-inference-scheduling/values-gaudi.yaml:
```diff
@@ -1,13 +1,21 @@
+#
+# Prerequisites:
+# - Intel Gaudi hardware (Gaudi 1, Gaudi 2, or Gaudi 3)
+# - Kubernetes 1.26+ with DRA feature gates enabled
```
> Is it a template for all the examples?

No. The kubeadm config is for DRA only, device plugins won't work with it.

yuanwu2017 committed:
Apply the httpRoute

Signed-off-by: yuanwu <yuan.wu@intel.com>

poussa reviewed (approved):
all good now.

yuanwu2017 committed:
Change the name from gaudi to hpu

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
Update the docs

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 commented:
@Gregory-Pereira Please help to review.

poussa commented on guides/inference-scheduling/README.md:
````diff
@@ -107,16 +108,21 @@ helmfile apply -e standalone -n ${NAMESPACE}
 ```
 
+
+**Note for Intel Gaudi (HPU) deployments:** Intel Gaudi requires Dynamic Resource Allocation (DRA) support. Ensure you have the [Intel Resource Drivers for Kubernetes](https://github.com/intel/intel-resource-drivers-for-kubernetes) installed on your cluster. See [Accelerator documentation](../../docs/accelerators/README.md#intel-gaudi-hpu) for setup details.
````
Intel Gaudi does not require but uses DRA. Reword like: "Intel Gaudi uses Dynamic Resource Allocation (DRA). Ensure..."

yuanwu2017 committed:
Fix typo

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
Update docs

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
Update values-hpu.yaml to use new accelerator config format

- Change from old 'dra' format to new 'accelerator' format per llm-d-modelservice chart
- Fix typo: hardware-vairant -> hardware-variant

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
Fix broken link fragment in README.md

Change #intel-gaudi-hpu to #intel-gaudi-hpu-configuration to match actual heading

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
docs: update accelerators README and fix broken link

- Remove redundant Intel Gaudi Configuration section (DRA section already has the link)
- Update link fragment from #intel-gaudi-hpu-configuration to #dynamic-resource-allocation

Signed-off-by: yuanwu <yuan.wu@intel.com>

yuanwu2017 committed:
Merge branch 'main' into hpu-ci

Signed-off-by: Yuan Wu <yuan.wu@intel.com>

Gregory-Pereira reviewed (approved):
/lgtm
